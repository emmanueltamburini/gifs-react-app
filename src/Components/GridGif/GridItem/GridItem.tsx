import { Gif } from "../../../interfaces/Gif.interfaces"

interface Props {
    image: Gif
}

export const GridItem = ({image}: Props) => {
  return (
    <div className="card">
        <img src={image.url} alt={image.title} />
        <p>{image.title}</p>
    </div>
  )
}
