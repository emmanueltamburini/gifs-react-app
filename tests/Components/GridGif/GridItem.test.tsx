import React from "react";
import { render } from "@testing-library/react"
import { Gif } from "../../../src/interfaces/Gif.interfaces";
import { GridItem } from '../../../src/Components/GridGif/GridItem/GridItem';

describe('GridItem test', () => {
    const image: Gif = {
            id: "TS9pTA670S1ajXEiK1",
            title: "Ace Of Spades Poker GIF",
            url: "https://media2.giphy.com/media/TS9pTA670S1ajXEiK1/giphy.gif?cid=47a56d216o55wkkv9735xyvrthi3vd3g5rtrzamij06uueeu&ep=v1_gifs_search&rid=giphy.gif&ct=g",
        };
    test('should be equal to snapshot', () => {
        const {container} = render(<GridItem image={image} />);
        expect(container).toMatchSnapshot();
    });
})