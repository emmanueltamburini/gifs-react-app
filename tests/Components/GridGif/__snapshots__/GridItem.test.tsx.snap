// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem test should be equal to snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ace Of Spades Poker GIF"
      src="https://media2.giphy.com/media/TS9pTA670S1ajXEiK1/giphy.gif?cid=47a56d216o55wkkv9735xyvrthi3vd3g5rtrzamij06uueeu&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Ace Of Spades Poker GIF
    </p>
  </div>
</div>
`;
